src/platform_linux/shmem.h

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018-2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <sys/types.h>
+#include <sys/shm.h>
+
+typedef struct shmem_info shmem_info;
+
+platform_status
+platform_shmcreate(size_t                size,
+                   platform_heap_handle *heap_handle,
+                   platform_heap_id     *heap_id);
+
+void
+platform_shmdestroy(platform_heap_handle *heap_handle);
+
+/*
+ * void * = splinter_shm_alloc(platform_heap_id heap_id, size_t nbytes,
+ *                             const char * objname)
+ *
+ * Caller-macro to invoke lower-level allocator and to pass-down caller's
+ * context fields, which are printed for diagnostics under a traceflag.
+ */
+#define splinter_shm_alloc(heap_id, nbytes, objname)                           \
+   platform_shm_alloc(heap_id, nbytes, objname, func, file, lineno)
+
+void *
+platform_shm_alloc(platform_heap_id hid,
+                   const size_t     size,
+                   const char      *objname,
+                   const char      *func,
+                   const char      *file,
+                   const int        lineno);
+
+/*
+ * void = splinter_shm_free(platform_heap_id heap_id, void *ptr,
+ *                          const char * objname)
+ *
+ * Caller-macro to invoke lower-level free method and to pass-down caller's
+ * context fields, which are printed for diagnostics under a traceflag.
+ */
+#define splinter_shm_free(heap_id, ptr, objname)                               \
+   platform_shm_free(heap_id, ptr, objname, func, file, lineno)
```

Comment:
Ignore this kind of wrapper caller-macros that you will see in part-1 and part-2.

I've streamlined the interfaces and call-sequences in part-3. We can review all that later and finalie.